### ISSUE TITLE
Returning objects with additional fields as search result

### ISSUE THREAD

**Issue by bal1anD** [weixsong/elasticlunr.js#82]
Hi,

I'm using elasticlunr for a project. I'm looking for the following feature:

Given documents of the following type:

```
[
   {
      text: " Hello world",
      id: 1
   },
  {
      text: "Amazing world",
      id:2
   },
........

]

```
If I do a search for, say val1

index.search("world");

I would like the result to be a list of documents which match such as:

`
[
   {
      text: " Hello world",
      id: 1
   },
  {
      text: "Amazing world",
      id:2
   }
]

`

I didn't see any documentation to achieve this in the current version.

**Comment by ToLuSt** [comment]
I don't know if I understand you right.
With the given code, you get references and the score to the matching documents.
And what you want to have is the corresponding document, right?

`index.saveDocument(true);`
This saves the documents.

`
const result = [];
index.search(query, {
            fields: {
                text: { boost: 2 },
            },
            bool: 'OR',
            expand: true
        })
            .map(({ ref, score }) => {
                // Get doc by ref
                const doc = index.documentStore.getDoc(reference);
                const obj = {text: doc.text, id: doc:id};
                results.push(obj);
            });
`
I think that should do the trick, if I didn't forget a thing.

**Comment by bal1anD** [comment]
Yeah this is what I want to achieve. I was wondering instead of doing the loop if elasticlunr has some efficient API to do this. I can use this functionality for now, feel free to resolve this issue. Thanks.

**Comment by ajorlando** [comment]
> Yeah this is what I want to achieve. I was wondering instead of doing the loop if elasticlunr has some efficient API to do this. I can use this functionality for now, feel free to resolve this issue. Thanks.

I'm also very interested to find out an answer to this. `index.search()`, accessed in dev tools on the homepage, returns an object reference by default ({savedDocument: true}) but I can't achieve this behavior locally or find anything documentation that speaks to this functionality.
